### PR TITLE
Individual cell voltages and batt_smbus

### PR DESCRIPTION
**Pull request by zselgrath**
Hello all,
This is my first pull request ever. I am still very new to development in general, so please forgive me. I know there are errors, But I would love some feedback while I fix them.

Changes to files are for two reasons:
- Try to fix an issue where the batt_smbus driver does not include I2C bus options for Pixhawk hardware with buses other than `PX4_I2C_BUS_ONBOARD` and `PX4_I2C_BUS_EXPANSION`  (Example: I am using a Pixhawk 4, so px4fmu-v5 defines the external I2C A port as `PX4_I2C_BUS_EXPANSION2`, which the driver cannot start on currently)
- Implement real individual voltage checking and reporting to the ground station via the existing individual voltage array in MAVlink `BATTERY_STATUS` messages. The existing code divides total pack voltage by number of cells (up to 10 cells) to calculate individual cell voltages, but I want to populate that array with measured cell voltages. 

The registers beyond 0x3f are specific to the custom voltage and temperature measuring board I am working on, but because they are not defined for most TI fuel gauge chips (the BQ40Z50 does not define any registers in the range of 0x40 - 0x49), I don't think I am breaking compatibility by adding this behavior. However, I do not have a BQ40Z50 to test this on.

Possible alternatives:
- Write custom driver for up to 10 cell voltages and up to three temperature values that works with my hardware to publish `battery_status` messages 
- Separate core SMbus functionality (`read_word, write_word, block_read, etc.`) from device specific functionality to prevent bloating of `batt_smbus` and make development easier as @dakejahl suggested 

Example:
Test bench setup for custom board connected to Pixhawk 4 I2C A port
img_20181005_200912

Starting the driver manually
screenshot from 2018-10-05 20-12-58

Result of sweeping benchtop through voltage range 
screenshot from 2018-10-05 20-15-06

Thanks,
Zack Selgrath

### CODE REVIEW

**Comment by mcsauder** [comment]
@AlexKlimaj

File: src/drivers/boards/px4fmu-v5/manifest.c [hunk 1]
```diff
@@ -66,15 +66,17 @@ typedef px4_hw_mft_list_entry_t *px4_hw_mft_list_entry;
...
-// The index of those components is given by the enum (px4_hw_mft_item_id_t)
-// declared in board_common.h
 static const px4_hw_mft_item_t hw_mft_list_v0500[] = {
 	{
 		.present     = 1,
 		.mandatory   = 1,
 		.connection  = px4_hw_con_onboard,
 	},
+	{
+		.present     = 1,
```
Comment (dagar) [inline src/drivers/boards/px4fmu-v5/manifest.c]: Is this change intentional?

**Comment by dagar** [comment]
Hi @zselgrath,

This looks good to me, although there's some ongoing batt_smbus refactoring work this should be coordinated with. https://github.com/PX4/Firmware/pull/10666 (@dakejahl )

The battery_status message is getting pretty overloaded. I think we'll need to split it for smart batteries and dumb analog sensors soon.

File: src/drivers/boards/px4fmu-v5/manifest.c (hunk 1 above)
Comment (zselgrath) [reply src/drivers/boards/px4fmu-v5/manifest.c]: No, it was not.

**Comment by zselgrath** [comment]
After talking with @dagar and reviewing @dakejahl 's current work on #10666 I have decided that its not really necessary to merge the changes specific to my hardware yet, but that I should open another pull request soon to merge the fixes for the driver.

**Comment by dagar** [comment]
Thanks @zselgrath, closing this PR as it will continue in several smaller pieces.
